File: learnsynth_offline_llm/android/src/main/kotlin/com/learnsynth/learnsynth_offline_llm/MlcRunner.kt
package com.learnsynth.learnsynth_offline_llm

class MlcRunner : Runner {
    override suspend fun load(modelPath: String) {
        TODO("Implement MLC runner")
    }

    override fun isReady(): Boolean = false

    override fun generateStream(
        prompt: String,
        maxTokens: Int,
        temperature: Double,
        onToken: (String) -> Unit
    ) {
        // TODO
    }

    override fun cancel() {
        // TODO
    }
}

File: learnsynth_offline_llm/ios/Classes/MlcRunner.swift
import Foundation

class MlcRunner: Runner {
    func load(modelPath: String) async {
        // TODO: implement real runner
    }

    func isReady() -> Bool { return false }

    func generateStream(prompt: String, maxTokens: Int, temperature: Double, onToken: @escaping (String) -> Void) {
        // TODO
    }

    func cancel() {
        // TODO
    }
}
